src/Fabulous.Avalonia/Views/Controls/AutoCompleteBox.fs

```diff
@@ -144,6 +162,13 @@ type AutoCompleteBoxModifiers =
     static member inline itemFilter(this: WidgetBuilder<'msg, #IFabAutoCompleteBox>, fn: string -> obj -> bool) =
         this.AddScalar(AutoCompleteBox.ItemFilter.WithValue(fn))
 
+    /// <summary>Sets the ItemTemplate property.</summary>
+    /// <param name="this">Current widget.</param>
+    /// <param name="template">The template to render the items with.</param>
+    [<Extension>]
+    static member inline itemTemplate(this: WidgetBuilder<'msg, #IFabAutoCompleteBox>, template: 'item -> WidgetBuilder<'msg, 'widget>) =
+        this.AddScalar(AutoCompleteBox.ItemTemplate.WithValue(WidgetHelpers.compileTemplate template))
```

Comment:
Is this a good place to compile the builder (here named `template`) into a template? Or do I lose the ability to identify different builders for the equality comparison by doing so? I'm doing it here because I was unable to declare the `ItemTemplate` attribute value with the open generic signature of the builder.

Reply:
No here. This is already done on `WidgetHelpers.buildItems`  as part of the WidgetBuilder constructor

Reply:
In a second look. It is ok to have `compileTemplate` as a separate function.

- `template: 'item -> WidgetBuilder<'msg, 'widget>` will be passing  `'item ` and then will be unboxed when `WidgetDataTemplate` is building the template.

- One thing we can do is to use `#IFabControl` instead of `'widget`